env/env.go

```diff
@@ -366,18 +368,39 @@ func Exec(s string) (string, error) {
 	return string(buf), nil
 }
 
-var cleanDoubleRE = regexp.MustCompile(`''`)
+var cleanDoubleRE = regexp.MustCompile(`(^|[^\\])''`)
 
 // Dequote unquotes a string.
-func Dequote(s string, c byte) (string, error) {
-	if len(s) < 2 || s[len(s)-1] != c {
+func Dequote(s string, quote byte) (string, error) {
+	if len(s) < 2 || s[len(s)-1] != quote {
 		return "", text.ErrUnterminatedQuotedString
 	}
 	s = s[1 : len(s)-1]
-	if c != '\'' {
-		return s, nil
+
```

Comment:
Not that long ago, I decided it's better to not include empty lines in Go code:

1. If code requires a visual break, it's better to fill that with an explanatory comment
2. Different individuals will occasionally mess up and put in multiple lines, not format their code, etc., or one person will introduce a line, the next will remove it, and so on. It's not that the people are arguing, but that they just may naturally do that. Since it's impossible to control this from individual to individual due to the impossibility of creating a standard guide on "when" and "where" an empty line to break up logic, I've come to the conclusion that it's just better to not have them anywhere. When/if a person decides it's necessary, they should see 1 
3. Less empty lines means its easier to view more lines at once on screen
4. Reduces commit "chatter" (see 2)

Reply:
I disagree, but I can commit to it. I've noticed that in your commits a while ago but I didn't know the reasons. Please put this info in `CONTRIBUTING.md` so we can reference it in other PRs.

Here are my opinions, but this is just FYI, we don't need to discuss any further:
1. This is subjective, but I don't think a comment is enough of a visual break.
2. Agree here, we can make such rules in `CONTRIBUTING.md`. BTW formatting code using `go fmt` is/should be a hard requirement for any Go project. Also, it doesn't allow more than one empty line :-)
3. Disagree, but this is subjective. I believe you only need the current context to fit on the screen, and this is usually a single function. If it's too big, break it down into smaller functions. Blocks separated with empty lines are natural candidates to be extracted (and named). But as always, "it depends". I guess it's not mentioned in either [Effective Go](https://golang.org/doc/effective_go.html) or [code review comments](https://github.com/golang/go/wiki/CodeReviewComments) since it's so subjective.

Reply:
Eh, it's not something worth spending a millisecond on. I just have a preference now for spaceless code. The reason I got to this is after working on a lot of other projects and seeing developers add/remove so many empty lines, I've come to the conclusion that the empty lines in blocks of code are worthless. The rationale/thinking kind of goes like this: if a block of code is more than 10-15 lines, it "should" be broken up into multiple functions (which I try to follow), and an empty line isn't really enough of a visual break. When code can't be broken up into multiple functions (such as large switch's or something else), and it's more than 10-15 lines, then each "block" should be delineated by a comment. Having _both_ an empty line and a comment is redundant. Again, this isn't worth debating / arguing over -- I'm not going to reject code with superfluous empty lines, I'm just stating that I think it's easier to read/view without the lines, especially when reviewing via patch sets (ie, in Git or some other patch based commit tracker).